### PR TITLE
Fix #18064, check illegal value

### LINKED ISSUE #18064: Scaling IncrementTask not handle BigDecimal correct
## Bug Report

When insert table data that not set the value of numeric column.
Scaling IncrementTask `PostgreSQLWalDumper` handle numeric to `BigDecimal` not corrently.

### Which version of ShardingSphere did you use?

master a159040ed39eb26838fc236f1e99c8ff22c4f336

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

Proxy

### Expected behavior

image

### Actual behavior

`new BigDecimal("")` or `new BigDecimal("null")` will cause exception.

### Reason analyze (If you can)

image

seem other data type will also cause problem.
```
[ERROR] 2022-05-30 16:55:05.771 [ShardingSphere-Scaling-Incremental-0130317c30317c3054317c7368617264696e675f6462-4] o.a.s.d.p.s.r.RuleAlteredJobScheduler - Incremental task execute failed.
java.lang.NumberFormatException: For input string: "null"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Long.parseLong(Long.java:589)
	at java.lang.Long.parseLong(Long.java:631)
	at org.apache.shardingsphere.data.pipeline.postgresql.ingest.wal.decode.TestDecodingPlugin.readColumnData(TestDecodingPlugin.java:170)
	at org.apache.shardingsphere.data.pipeline.postgresql.ingest.wal.decode.TestDecodingPlugin.readColumn(TestDecodingPlugin.java:129)
	at org.apache.shardingsphere.data.pipeline.postgresql.ingest.wal.decode.TestDecodingPlugin.readWriteRowEvent(TestDecodingPlugin.java:90)
	at org.apache.shardingsphere.data.pipeline.postgresql.ingest.wal.decode.TestDecodingPlugin.readTableEvent(TestDecodingPlugin.java:69)
	at org.apache.shardingsphere.data.pipeline.postgresql.ingest.wal.decode.TestDecodingPlugin.decode(TestDecodingPlugin.java:51)
	at org.apache.shardingsphere.data.pipeline.postgresql.ingest.PostgreSQLWalDumper.dump(PostgreSQLWalDumper.java:91)
	at org.apache.shardingsphere.data.pipeline.postgresql.ingest.PostgreSQLWalDumper.doStart(PostgreSQLWalDumper.java:75)
	at org.apache.shardingsphere.data.pipeline.api.executor.AbstractLifecycleExecutor.start(AbstractLifecycleExecutor.java:41)
	at org.apache.shardingsphere.data.pipeline.core.task.IncrementalTask.doStart(IncrementalTask.java:85)
	at org.apache.shardingsphere.data.pipeline.api.executor.AbstractLifecycleExecutor.start(AbstractLifecycleExecutor.java:41)
	at org.apache.shardingsphere.data.pipeline.api.executor.AbstractLifecycleExecutor.run(AbstractLifecycleExecutor.java:61)
	at java.util.concurrent.CompletableFuture$AsyncRun.run$$$capture(CompletableFuture.java:1640)
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
```

### PR DESCRIPTION
**Pull request by azexcy**
Fix issue #18064
Changes proposed in this pull request:
- check illegal value

### CODE REVIEW

**Comment by azexcy** [comment]
PostgerSQL TextDecoder:
```
table public.t_order_1: INSERT: order_id[bigint]:11 user_id[integer]:1 status[character varying]:'ok' timestp[timestamp without time zone]:null
```

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-postgresql/src/main/java/org/apache/shardingsphere/data/pipeline/postgresql/ingest/wal/decode/TestDecodingPlugin.java [hunk 1]
```diff
@@ -153,36 +155,46 @@ private String readColumnType(final ByteBuffer data) {
     }
     
     private Object readColumnData(final ByteBuffer data, final String columnType) {
-        if (columnType.startsWith("numeric")) {
-            return new BigDecimal(readNextSegment(data));
-        }
         if (columnType.startsWith("bit") || columnType.startsWith("bit varying")) {
             return readNextSegment(data);
         }
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-postgresql/src/main/java/org/apache/shardingsphere/data/pipeline/postgresql/ingest/wal/decode/TestDecodingPlugin.java:160]: Does "bit" and "bit varying" need to do the same change?

Comment (azexcy) [reply shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-postgresql/src/main/java/org/apache/shardingsphere/data/pipeline/postgresql/ingest/wal/decode/TestDecodingPlugin.java:160]: Yes, I will fix it.

**Comment by azexcy** [comment]
Closed,because #18084 fixed same problem

**Pull request closed without merging**